test/testAdaptors/DSRAdaptor.t.sol

```diff
@@ -0,0 +1,215 @@
+
+        cellar.addAdaptorToCatalogue(address(dsrAdaptor));
+        cellar.addPositionToCatalogue(dsrPosition);
+
+        cellar.addPosition(0, dsrPosition, abi.encode(0), false);
+        cellar.setHoldingPosition(dsrPosition);
+
+        DAI.safeApprove(address(cellar), type(uint256).max);
+
+        initialAssets = cellar.totalAssets();
+    }
+
+    function testDeposit(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(DAI), address(this), assets);
+        cellar.deposit(assets, address(this));
+
+        assertApproxEqAbs(
+            cellar.totalAssets(),
+            initialAssets + assets,
+            2,
+            "Cellar totalAssets should equal assets + initial assets"
+        );
+    }
+
+    function testWithdraw(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(DAI), address(this), assets);
+        cellar.deposit(assets, address(this));
+
+        uint256 maxRedeem = cellar.maxRedeem(address(this));
+
+        cellar.redeem(maxRedeem, address(this), address(this));
+    }
+
+    function testInterestAccrual(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(DAI), address(this), assets);
+        cellar.deposit(assets, address(this));
+
+        console.log("TA", cellar.totalAssets());
+
+        uint256 assetsBefore = cellar.totalAssets();
+
+        vm.warp(block.timestamp + 1 days);
+        mockDaiUsd.setMockUpdatedAt(block.timestamp);
+
+        assertEq(
+            cellar.totalAssets(),
+            assetsBefore,
+            "Assets should not have increased because nothing has interacted with dsr."
+        );
+
+        uint256 bal = manager.daiBalance(address(cellar));
+        assertGt(bal, assets, "Balance should have increased.");
+
+        uint256 assetsAfter = cellar.totalAssets();
+
+        assertGt(assetsAfter, assetsBefore, "Total Assets should have increased.");
+    }
+
+    function testUsersDoNotGetPendingInterest(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(DAI), address(this), assets);
+        cellar.deposit(assets, address(this));
+
+        uint256 assetsBefore = cellar.totalAssets();
+
+        vm.warp(block.timestamp + 1 days);
+        mockDaiUsd.setMockUpdatedAt(block.timestamp);
+
+        assertEq(
+            cellar.totalAssets(),
+            assetsBefore,
+            "Assets should not have increased because nothing has interacted with dsr."
+        );
+
+        uint256 maxRedeem = cellar.maxRedeem(address(this));
+        cellar.redeem(maxRedeem, address(this), address(this));
+
+        assertApproxEqAbs(DAI.balanceOf(address(this)), assets, 3, "Should have sent DAI to the user.");
+
+        uint256 bal = manager.daiBalance(address(cellar));
+        assertGt(bal, 0, "Balance should have left pending yield in DSR.");
```

Comment:
Do we not want to pull accumulated yield as well from the DSR when carrying out a `redeem` w/ `maxRedeem` amounts?

Reply:
We talked about this a bit in the meeting, but please elaborate. In general, it looks like the `dsrManager` exit call updates `chi` and thus the total exited amount has interest included. So wouldn't the `cellar` have the extra interest?

Reply:
No `exit` will only pull out the amount of assets you specify, so the pending yield is left in the DSR, which is fine, the strategist can handle that however they want to during a rebalance.

Reply:
Roger that, looks good then! Thanks for the clarification.